[EuiDescriptionList] New `gutterSize` prop

breehall opened the pull request:
### Summary
Fixes #5834 

Creates the `gutterSize` prop for `EuiDescriptionList` which is responsible for adjusting vertical spacing between Description List elements. `gutterSize` accepts the t-shirt sizes `s` and `m` (default) to add a top margin to Description List elements. When no `gutterSize` is provided, size `m` is used by default.

**Small `gutterSize`**
<img width="956" alt="image" src="https://user-images.githubusercontent.com/40739624/186504818-575d61d1-6b17-436b-b0b9-859746c77bb6.png">
<hr>

**Medium `gutterSize`**
<img width="964" alt="image" src="https://user-images.githubusercontent.com/40739624/186504903-e1d52dd7-4342-4c80-a081-a400b00ca8c4.png">

### Checklist

- [x] Checked in both **light and dark** modes
- [x] Checked in **mobile**
- [x] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
- [x] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#adding-playground-toggles)**
- [x] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md)**
- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for any docs examples
- [x] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/cypress-testing.md) tests**
- [x] Checked for **breaking changes** and labeled appropriately
- [x] Checked for **accessibility** including keyboard-only and screenreader modes
- [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_6175/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_6175/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_6175/

thompsongl commented on src/components/description_list/description_list_description.tsx:
```diff
@@ -47,6 +47,9 @@ export const EuiDescriptionListDescription: FunctionComponent<EuiDescriptionList
       if (align === 'center') {
         conditionalStyles.push(styles.left);
       }
+      gutterSize === 's'
+        ? conditionalStyles.push(styles.s)
+        : conditionalStyles.push(styles.m);
```
```suggestion
      conditionalStyles.push(styles[gutterSize]);
```

thompsongl reviewed (approved):
LGTM!
Just one suggestion

constancecchen commented:
2 thoughts:

1. Should `gutterSize` also affect the left/right padding on `type="column" align="center"` description lists?
<img width="359" alt="Screen Shot 2022-08-25 at 9 29 40 AM" src="https://user-images.githubusercontent.com/549407/186720196-6868e037-13de-4bc4-a4e7-f0b94bea719d.png">

2. Should `compressed` DescriptionLists default to a gutterSize of `s` instead of `m`? If not / if we're OK with making people manually do `compressed gutterSize="s"`, I think we should document `gutterSize` in the `Centered and compressed` instead of creating a new section for it. It feels like it generally belongs in the concept of "further tailor[ing] the look of a description list" and does not need its own section.

breehall commented:
> Should gutterSize also affect the left/right padding on type="column" align="center" description lists?

I don't believe so because the prop is intended to be for vertical spacing only, but may I ask why it should?

> Should compressed DescriptionLists default to a gutterSize of s instead of m?

Possibly, but I think it would affect current iterations of `DescriptionList`. I think I'd be ok with allowing consumers to make that decision on their own instead of us making it for them. But I'll wait for Elizabet to chime in.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_6175/

constancecchen commented:
> I don't believe so because the prop is intended to be for vertical spacing only, but may I ask why it should?

That's not my personal intepretation of  "gutter" - when I read "gutterSize", I would assume it applies the sizing to both horizontal and vertical gutters, not just vertical. That's how `EuiFlexGroup` works, for example. If we want to be extremely specific and only apply this change to vertical gutters, then we should name the prop accordingly, e.g. `rowGutterSize`.

> I think I'd be ok with allowing consumers to make that decision on their own instead of us making it for them.

Same, but in that case, let's combine the `Centered and compressed` section/example with the new gutter example and name it something more generic `Customizing appearance`. It might also make sense to add toggles for all the props being demo'd in that section (`compressed`, `align`, and `gutterSize`).

breehall commented:
Ok, gotcha. So as far as current take aways:
1. Update `gutterSize` prop name to be more specific (like `rowGutterSize)
2. Combine `DescriptionList` appearance modifiers into one example

constancecchen commented:
I'd hang tight on 1. to see if @miukimiu thinks we're OK to apply the size change to the horizontal spacing as well as vertical, but you should be good to move ahead on 2.!

breehall commented:
I updated the Centered & Compressed example to be Customizing Appearance. I've also added toggles for the `align`, `compressed`, and `gutterSize` props.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_6175/

constancecchen reviewed (approved):
🎉 🎉 Love the UX for the new docs!! I'll open a follow-up issue for @miukimiu to consider when she gets back about `gutterSize` and the horizontal gutter.
